www/js/maslow.js

```diff
@@ -0,0 +1,208 @@
+// When we can change to proper ESM - uncomment this
+// import M from "constants";
+
+/** Maslow Status */
+let maslowStatus = { homed: false, extended: false };
+
+/** This keeps track of when we saw the last heartbeat from the machine */
+let lastHeartBeatTime = new Date().getTime();
+
+const err = "error: ";
+// When we can change to proper ESM - prefix these const strings and functions with 'export' (minus the quotes of course)
+const MaslowErrMsgKeyValueCantUse = `${err}Could not use supplied key-value pair.`;
+const MaslowErrMsgNoKey = `${err}No key supplied for value.`;
+const MaslowErrMsgNoValue = `${err}No value supplied for key.`;
+const MaslowErrMsgNoMatchingKey = `${err}Could not find key for value in reference table.`;
+const MaslowErrMsgKeyValueSuffix = "This is probably a programming error\nKey-Value pair supplied was:";
+
+/** Perform maslow specific-ish info message handling */
+const maslowInfoMsgHandling = (msg) => {
+    if (msg.startsWith('MINFO: ')) {
+        maslowStatus = JSON.parse(msg.substring(7));
+        return true;
+    }
+
+    if (msg.startsWith('[MSG:INFO: Heartbeat')) {
+        lastHeartBeatTime = new Date().getTime();
+        return true;
+    }
+
+    //Catch the calibration complete message and alert the user
+    if (msg.startsWith('[MSG:INFO: Calibration complete')) {
+        alert('Calibration complete. You do not need to do calibration ever again unless your frame changes size. You might want to store a backup of your maslow.yaml file in case you need to restore it later.');
+        return true;
+    }
+
+    return false;
+}
+
+/** Perform maslow specific-ish error message handling */
+const maslowErrorMsgHandling = (msg) => {
+    if (!msg.startsWith("error:")) {
+        // Nothing to see here - move along
```

Comment:
At first I thought this comment was for the reader, as in "don't worry about this code, just keep reading 😛